File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        


        # array of prices given
        # max profit by choosing a single day to buy 1 stock and choose next day in future to sell
        if not prices: return 0 

        profit = 0
        bought = prices[0]
        for price in prices:
            profit = max(profit, price - bought)
            bought = min(bought, price)

        return profit

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int maxProfitAns = 0;
        int buyPrice = Integer.MAX_VALUE;
        for (int price: prices){
            if (buyPrice > price){
                buyPrice = price;
            }
            maxProfitAns = Math.max(maxProfitAns, price - buyPrice);
        }
        return maxProfitAns;
    }
}
